src/raps/actions/swap.ts

```diff
@@ -17,9 +18,15 @@ import store from '@/redux/store';
 import { greaterThan } from '@/helpers/utilities';
 import { AllowancesCache, ethereumUtils, gasUtils } from '@/utils';
 import logger from '@/utils/logger';
+import { MMKV } from 'react-native-mmkv';
+import { STORAGE_IDS } from '@/model/mmkv';
 
 const actionName = 'swap';
 
+export const metadataStorage = new MMKV({
+  id: STORAGE_IDS.SWAPS_METADATA_STORAGE,
+});
```

Comment:
I think I can, but I am hesititing before creating ad-hoc stardards. This is not stardarized in other places and in my understanding is that we should first discuss it and then decide. 
Also, I think I share a slightly different opinion. Firstly, `handlers` were designed to deal with external clients, storage is technically not a client (I know, we break this rule). Additionally, I am more a fan of "feature-based" separation, not one based on technicalities.
Seperation under "hooks", "storage", and "components" is not very beneficial and makes it difficult to yield ownership over anything, when every feature becomes granularized. That's the reason why I typically put hook next to component (instead of "hooks" dir) and also this is the reason why I put the storage here. I know, this calls for refactoring, when we'll have a better vision, but I don't want to proceed (and in this case, even into a pattern we already find not practical